src/main/webapp/app/trial/trial.component.html

```diff
@@ -81,28 +82,39 @@
                     <label for="retrieve" class="noteAdd">No </label>
                     <input id="retrieve" type="radio" value="No" name="archive" [(ngModel)]="trialChosen.archived" (change)="updateStatus('archive')">
                 </span>
-            <br/><br/>
+        </div>
+        <div>
+            <label for="last-update">Protocol Accessed on: </label>
+            <span id="last-update" class="noteEdit"> {{ trialChosen.protocol_accessed | date: 'yyyy-MM-dd' }}
+                <button *ngIf="isPermitted" class="btn btn-primary noteEdit" (click)="updateProtocolAccessedDate()">Update</button>
+                <span *ngIf="protocolAccessedMessage.content" class="load-span" [style.color]="protocolAccessedMessage.color"> {{protocolAccessedMessage.content}} </span>
+            </span>
+        </div>
+        <div class="trial-margin">
             <label for="nctId">NCT ID: </label>
             <span id="nctId" class="noteEdit">
                 <a target="_blank" href="https://clinicaltrials.gov/ct2/show/{{trialChosen.nct_id}}">{{ trialChosen.nct_id }}</a>
             </span>
-            <br/>
+        </div>
+        <div>
             <label for="protocolNo">Protocol No: </label>
             <span id="protocolNo" class="noteEdit">
-                <input [(ngModel)]="trialChosen.protocol_no" (blur)="updateProtocolNo()" (keypress)="clearMessage('protocol_no')"/>
+                <input [(ngModel)]="trialChosen.protocol_no" (blur)="updateProtocolNo()" (keypress)="clearMessageByName('protocolNo')"/>
```

Comment:
These  keys should be put into constants if possible.

Reply:
And I think most of places in this project, you use underscore instead of camelCase? Why not keep that?

Reply:
Underscore is used for data model attributes, which is follow the match-engine naming rule. Functions still use camelCase.